google/cloud/pubsub/internal/subscriber_connection_impl.cc

```diff
@@ -53,31 +55,51 @@ StatusOr<pubsub::PullResponse> SubscriberConnectionImpl::Pull() {
 
   google::pubsub::v1::PullRequest request;
   request.set_subscription(subscription.FullName());
+  // Pubsub returns at most 1 message.
   request.set_max_messages(1);
 
-  auto response = google::cloud::internal::RetryLoop(
-      current.get<pubsub::RetryPolicyOption>()->clone(),
-      current.get<pubsub::BackoffPolicyOption>()->clone(),
-      google::cloud::Idempotency::kIdempotent,
-      [stub = stub_](auto& context, auto const& request) {
-        return stub->Pull(context, request);
-      },
-      request, __func__);
-  if (!response) return std::move(response).status();
-  if (response->received_messages_size() != 1) {
-    return internal::InternalError("invalid response, mismatched ID count",
-                                   GCP_ERROR_INFO());
+  auto retry_policy = current.get<pubsub::RetryPolicyOption>()->clone();
+  auto backoff_policy = current.get<pubsub::BackoffPolicyOption>()->clone();
+  google::cloud::Status last_status;
+  google::cloud::StatusOr<google::pubsub::v1::PullResponse> response;
```

Comment:
`response` is only used inside the loop, so maybe move it there and you can use `auto response = ...`.